packages/commonwealth/client/scripts/controllers/server/sessions.ts

```diff
@@ -0,0 +1,24 @@
+import { ethers } from 'ethers';
+
+class SessionsController {
+  wallets: Record<string, ethers.Wallet>;
+  // TODO: store expiration time on wallets[]
+
+  constructor() {
+    this.wallets = {}
+  }
+
+  public getAddress(chainId: number): string {
+    return this.wallets[chainId]?.address;
+  }
+
+  public getOrCreateAddress(chainId: number): string {
+    if (this.wallets[chainId]) {
+      return this.wallets[chainId].address;
+    }
+    this.wallets[chainId] = ethers.Wallet.createRandom();
+    return this.wallets[chainId].address;
+  }
+}
+
+export default SessionsController;
```

Comment:
How is this going to be extended for other networks? (cosmos, substrate, etc)

Reply:
Check out my session keys 2 branch, also

https://polkadot.js.org/docs/api/start/keyring/